lib/smart_properties.rb

```diff
@@ -108,13 +108,15 @@ def included(base)
   # @param [Hash] attrs the set of attributes that is used for initialization
   #
   def initialize(*args, &block)
+    properties = self.class.properties.to_hash
```

Comment:
Ok, this might be a tough one: Why do you need a new instance of the properties? The following code never mutates the properties hash itself, so it shouldn't be necessary. Am I missing something?